tools/javadoc/javadoc.bzl

```diff
@@ -81,13 +83,13 @@ def _javadoc_library(ctx):
 
     # TODO(ronshapiro): Should we be using a different tool that doesn't include
     # timestamp info?
-    jar_command = "%s/bin/jar cf %s -C tmp ." % (java_home, ctx.outputs.jar.path)
+    jar_command = "%s/bin/jar cf %s -C %s ." % (java_home, ctx.outputs.jar.path, tmp.path)
 
     srcs = depset(transitive = [src.files for src in ctx.attr.srcs]).to_list()
     ctx.actions.run_shell(
         inputs = srcs + classpath + ctx.files._jdk,
         command = "%s && %s" % (" ".join(javadoc_command), jar_command),
-        outputs = [ctx.outputs.jar],
+        outputs = [tmp, ctx.outputs.jar],
```

Comment:
I am fairly inexperienced with bazel, but without that change it is not allowed to call define_directory at all. The variable name was perhaps badly chosen, I picked it just in the process of refactoring out the directory `tmp` from the `-d` argument.

But it does make sense that bazel wants to know which actual action produces the directory so that it be aware of changes, when it should be deleted before re-running the action, etc, so I suspect it must be declared here. I haven't found an example of a task that declares a directory and deletes it before being finished to see if that would be "correct" (my concern is that it wouldn't be correct, but might appear to work in some situations...).

Reply:
Interesting.

The variable name is okay, although I guess you could call it `output_dir` or something like that if you want.

Reply:
Sure, I'll make that change.